cypress/integration/console/shared/overview/overview.spec.js

```diff
@@ -0,0 +1,208 @@
+      name: 'Gateway Test Name',
+      description: 'Gateway Test Description',
+      gateway_server_address: 'test-address',
+      frequency_plan_id: 'EU_863_870',
+    }
+
+    before(() => {
+      cy.loginConsole({ user_id: userId, password: user.password })
+      cy.createGateway(gateway, userId)
+      cy.clearLocalStorage()
+      cy.clearCookies()
+    })
+
+    beforeEach(() => {
+      cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+    })
+
+    it('displays UI elements in place', () => {
+      cy.visit(`${Cypress.config('consoleRootPath')}/gateways/${gatewayId}`)
+
+      cy.findByText(`${gateway.name}`, { selector: 'h1' }).should('be.visible')
+      cy.findByRole('link', { name: /Collaborator/ }).should('be.visible')
+      cy.findByRole('link', { name: /API key/ }).should('be.visible')
+
+      cy.findByRole('row', { name: new RegExp(gatewayId) }).should('be.visible')
+      cy.findByText(new RegExp(gateway.ids.eui)).should('be.visible')
+      cy.findByText(new RegExp(gateway.description)).should('be.visible')
+      cy.findByText(new RegExp(gateway.gateway_server_address)).should('be.visible')
+      cy.findByText(new RegExp(gateway.frequency_plan_id)).should('be.visible')
+
+      cy.findByTestId('events-widget').should('be.visible')
+      cy.findByTestId('map-widget').should('be.visible')
+
+      cy.findByTestId('error-notification').should('not.exist')
+    })
+  })
+
+  describe('Organization', () => {
+    const organizationId = 'org-overview-test'
+    const organization = {
+      ids: { organization_id: organizationId },
+      name: 'Organization Test Name',
+    }
+
+    before(() => {
+      cy.loginConsole({ user_id: userId, password: user.password })
+      cy.createOrganization(organization, userId)
+      cy.clearLocalStorage()
+      cy.clearCookies()
+    })
+
+    beforeEach(() => {
+      cy.loginConsole({ user_id: user.ids.user_id, password: user.password })
+    })
+
+    it('displays UI elements in place', () => {
+      cy.visit(`${Cypress.config('consoleRootPath')}/organizations/${organizationId}`)
+
+      cy.findByText(`${organization.name}`, { selector: 'h1' }).should('be.visible')
+      cy.findByRole('link', { name: /Collaborator/ }).should('be.visible')
+      cy.findByRole('link', { name: /API key/ }).should('be.visible')
+
+      cy.findByRole('row', { name: new RegExp(organizationId) }).should('be.visible')
+
+      cy.findByTestId('events-widget').should('be.visible')
+
+      cy.findByTestId('error-notification').should('not.exist')
+    })
+  })
+
+  describe('End Devices', () => {
+    const applicationId = 'app-end-devices-overview-test'
+    const application = {
+      ids: { application_id: applicationId },
+      name: 'Application End Devices Test Name',
+      description: `Application End Devices Test Description`,
+    }
+
+    const endDeviceId = 'end-device-overview-test'
+    const endDevice = {
```

Comment:
```suggestion
    const endDeviceRequestBody = {
```

Maybe cleaner to split this up:

```
const endDevice = { ... }
const endDeviceFieldMask = { ... }
const endDeviceRequestBody = {
  end_device: endDevice,
  field_mask: endDeviceFieldMask,
}
```

And maybe also worth thinking about using the SDK for these use cases. (For later concern)